evalml/pipelines/components/transformers/preprocessing/featuretools.py

```diff
@@ -27,7 +27,7 @@ def __init__(self, index="index", features=None, random_seed=0, **kwargs):
 
         self.index = index
         self.features = features
-        self._passed_in_features = True if features else None
+        self._passed_in_features = True if features is not None else None
```

Comment:
To make this fix, I needed the ability to set the parameters of the DFS Transformer so that a `transform` call would be no op. One option I considered was only passing in the `IdentityFeature` objects in at ` _handle_partial_dependence_fast_mode`, but I felt that left some ambiguity as to what the behavior would be. 

Therefore, I updated the behavior here to no longer treat an empty list passed in for `features` the same as `features` being unspecified. This change made sense to me, but I wanted to highlight it in case anyone disagrees or thinks this could be problematic.

Reply:
Maybe the warning could come from AutoMLSearch itself. If it sees that a user is trying to have a DFS Transformer in automl search with no features, it could raise that warning. If it's passed directly into a pipeline, maybe we're okay not warning users since it might be more clear to users at that point what the impact would be. 

Then again, my goal here was specifically to have no-op functionality in the DFS Transformer, but I definitely get the discomfort. At the very least, I need to update the dosctring making this explicit.

Reply:
Ah, the warning within AutoMLSearch is a good idea. Having it in the docstring is also a super good callout, sounds like a good path forward to me

Reply:
Turns out AutomlSearch doesn’t add the DFS Transformer to pipelines if features is an empty list.

We definitely add it to the internal search parameters [here](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_search.py#L736-L740) because we explicitly check if it’s None or not, but when we make_pipeline we exclude it if features is falsy: https://github.com/alteryx/evalml/blob/main/evalml/pipelines/utils.py#L412. We do the same for split pipelines. So currently in automl, an empty features list already is no-op.

My preferance here would be to not touch the automl code (though, maybe it’d be worth adding a note to the docstring there about the behavior when features is an empty list). And it makes me more confident that we probably want this behavior for the component itself as well.

Reply:
Yep, updated the automl docstring to make this clear there as well